### DISCUSSION TITLE
Document Symbols: remove details

### DISCUSSION THREAD

**Discussion by gipo355** [folke/trouble.nvim#561] [Q&A]
Hi everyone and folke!

As always, thanks for the amazing work!!

I am trying to setup the amazing new symbol functionality to have some kind of a minimap on the right side of the buffer.

It works wonderfully, it tracks my current position.

The only problem i'm facing is the details are making it a bit difficult to use.

Screenshot:

![image](https://github.com/user-attachments/assets/284d37a8-d5db-4f18-8a10-57498e44f761)

![image](https://github.com/user-attachments/assets/99710b43-f7c4-43aa-9a67-3a9946337d54)

![image](https://github.com/user-attachments/assets/e351491c-63cf-4a3a-bf19-57c57aa8eb85)

As can be seen, the details (comment highlight code) creates bigger funcs and leaves holes in the map.

is it possible to disable them?

**Comment by MKrings** [comment]
I had the same problem and struggled with it for several hours in the plugin's source code before stumbling upon [this comment](https://github.com/folke/trouble.nvim/issues/508#issuecomment-2168341714) in a Github issue. While the plugin itself is great, the documentation on how to change the formatters is basically non-existent (read: absolutely atrocious). The config snippet for this particular problem (bare symbol name without any other annotations) would look like this:

>{
>    "folke/trouble.nvim",
>    opts = {  modes = { lsp_document_symbols = { format = "{kind_icon} {symbol.name}" }}},
>},

Note that the above snippet is for a minimal Lazy.vim configuration. However, you can just copy-paste the `opt` line into your config file and it *should* work just fine. This will overwrite the [default formatter](https://github.com/folke/trouble.nvim/blob/6efc446226679fda0547c0fd6a7892fd5f5b15d8/lua/trouble/sources/lsp.lua#L60) for document symbols.
